train/utils.py

```diff
@@ -60,167 +58,169 @@ def to_tensor(array):
+            "FN": fn,
+            "TN": tn,
+            "Precision": precision,
+            "Recall": recall,
+            "F1": f1,
+            "Accuracy": accuracy,
+        }
+
+
+def plot_prediction_vs_ground_truth(
+    yo_pred, yp_pred, yn_pred, yo_true, yp_true, yn_true
+):
+    yo_pred_np = to_numpy(yo_pred)
+    yp_pred_np = to_numpy(yp_pred)
+    yn_pred_np = None if s.remove_yn else to_numpy(yn_pred)
 
-    yo_pred_np = to_numpy(F.sigmoid(yo_pred))
     yo_true_np = to_numpy(yo_true)
-    yp_pred_np = to_numpy(F.sigmoid(yp_pred))
-    if yn_pred is not None:
-        yn_pred_np = to_numpy(F.sigmoid(yn_pred))
-    yn_true_np = to_numpy(yn_true)
-
-    plt.figure(figsize=(12, 8))
-
-    plt.subplot(3, 2, 1)
-    plt.imshow(yo_true_np, aspect="auto", cmap="viridis")
-    plt.title("YO Ground Truth")
-    plt.colorbar()
-
-    plt.subplot(3, 2, 2)
-    plt.imshow(yo_pred_np, aspect="auto", cmap="viridis")
-    plt.title("YO Prediction")
-    plt.colorbar()
-
-    plt.subplot(3, 2, 3)
-    plt.imshow(yn_true_np, aspect="auto", cmap="inferno")
-    plt.title("YP Ground Truth")
-    plt.colorbar()
-
-    plt.subplot(3, 2, 4)
-    plt.imshow(yp_pred_np, aspect="auto", cmap="inferno")
-    plt.title("YP Prediction")
-    plt.colorbar()
-
-    if yn_pred is not None:
-        plt.subplot(3, 2, 5)
-        plt.imshow(yn_true_np, aspect="auto", cmap="inferno")
-        plt.title("YN Ground Truth")
-        plt.colorbar()
-
-        plt.subplot(3, 2, 6)
-        plt.imshow(yn_pred_np, aspect="auto", cmap="inferno")
-        plt.title("YN Prediction")
-        plt.colorbar()
+    yp_true_np = to_numpy(yp_true)
+    yn_true_np = None if s.remove_yn else to_numpy(yn_true)
+
+    # Numero di righe: 2 per YO e YP, +1 se consideri YN
+    n_rows = 2 if s.remove_yn else 3
+    n_cols = 2  # sempre 2 colonne: ground truth e predizione
+
+    fig, axes = plt.subplots(n_rows, n_cols, figsize=(12, 4 * n_rows))
+
+    # Plot YO
+    im0 = axes[0, 0].imshow(yo_true_np, aspect="auto", origin="lower")
+    axes[0, 0].set_title("YO Ground Truth")
+    fig.colorbar(im0, ax=axes[0, 0])
+
+    im1 = axes[0, 1].imshow(yo_pred_np, aspect="auto", origin="lower")
+    axes[0, 1].set_title("YO Prediction")
+    fig.colorbar(im1, ax=axes[0, 1])
+
+    # Plot YP
+    im2 = axes[1, 0].imshow(yp_true_np, aspect="auto", origin="lower")
+    axes[1, 0].set_title("YP Ground Truth")
+    fig.colorbar(im2, ax=axes[1, 0])
+
+    im3 = axes[1, 1].imshow(yp_pred_np, aspect="auto", origin="lower")
+    axes[1, 1].set_title("YP Prediction")
+    fig.colorbar(im3, ax=axes[1, 1])
+
+    # Plot YN (solo se non rimosso)
+    if not s.remove_yn:
+        im4 = axes[2, 0].imshow(yn_true_np, aspect="auto", origin="lower")
+        axes[2, 0].set_title("YN Ground Truth")
+        fig.colorbar(im4, ax=axes[2, 0])
+
+        im5 = axes[2, 1].imshow(yn_pred_np, aspect="auto", origin="lower")
+        axes[2, 1].set_title("YN Prediction")
+        fig.colorbar(im5, ax=axes[2, 1])
```

Comment:
Domanda, si riuscirebbe a fare in modo che la scala di colori vada sempre da 0 a 1, in modo da avere grafici coerenti nel tempo?
Forse si può fare qualcosa [tipo questo](https://www.reddit.com/r/learnpython/comments/17089oo/matplotlib_force_colorbar_to_show_range_that_is/).